src/main/java/org/openhab/automation/jrule/internal/module/RuleEventFactory.java

```diff
@@ -0,0 +1,188 @@
+        return new RuleRemovedEvent(topic, payload, source, ruleDTO);
+    }
+
+    private Event createRuleAddedEvent(String topic, String payload, @Nullable String source) {
+        RuleDTO ruleDTO = deserializePayload(payload, RuleDTO.class);
+        return new RuleAddedEvent(topic, payload, source, ruleDTO);
+    }
+
+    private String getRuleId(String topic) {
+        String[] topicElements = getTopicElements(topic);
+        if (topicElements.length != 4) {
+            throw new IllegalArgumentException("Event creation failed, invalid topic: " + topic);
+        }
+        return topicElements[2];
+    }
+
+    /**
+     * Creates a rule updated event.
+     *
+     * @param rule the new rule.
+     * @param oldRule the rule that has been updated.
+     * @param source the source of the event.
+     * @return {@link RuleUpdatedEvent} instance.
+     */
+    public static RuleUpdatedEvent createRuleUpdatedEvent(Rule rule, Rule oldRule, String source) {
+        String topic = buildTopic(RULE_UPDATED_EVENT_TOPIC, rule);
+        final RuleDTO ruleDto = RuleDTOMapper.map(rule);
+        final RuleDTO oldRuleDto = RuleDTOMapper.map(oldRule);
+        List<RuleDTO> rules = new LinkedList<>();
+        rules.add(ruleDto);
+        rules.add(oldRuleDto);
+        String payload = serializePayload(rules);
+        return new RuleUpdatedEvent(topic, payload, source, ruleDto, oldRuleDto);
+    }
+
+    /**
+     * Creates a rule status info event.
+     *
+     * @param statusInfo the status info of the event.
+     * @param ruleUID the UID of the rule for which the event is created.
+     * @param source the source of the event.
+     * @return {@link RuleStatusInfoEvent} instance.
+     */
+    public static RuleStatusInfoEvent createRuleStatusInfoEvent(RuleStatusInfo statusInfo, String ruleUID,
+            String source) {
+        String topic = buildTopic(RULE_STATE_EVENT_TOPIC, ruleUID);
+        String payload = serializePayload(statusInfo);
+        return new RuleStatusInfoEvent(topic, payload, source, statusInfo, ruleUID);
+    }
+
+    /**
+     * Creates a rule removed event.
+     *
+     * @param rule the rule for which this event is created.
+     * @param source the source of the event.
+     * @return {@link RuleRemovedEvent} instance.
+     */
+    public static RuleRemovedEvent createRuleRemovedEvent(Rule rule, String source) {
+        String topic = buildTopic(RULE_REMOVED_EVENT_TOPIC, rule);
+        final RuleDTO ruleDto = RuleDTOMapper.map(rule);
+        String payload = serializePayload(ruleDto);
+        return new RuleRemovedEvent(topic, payload, source, ruleDto);
+    }
+
+    /**
+     * Creates a rule added event.
+     *
+     * @param rule the rule for which this event is created.
+     * @param source the source of the event.
+     * @return {@link RuleAddedEvent} instance.
+     */
+    public static RuleAddedEvent createRuleAddedEvent(Rule rule, String source) {
```

Comment:
Also unused. Are we missing the code that was supposed to use this method, or is maybe this PR still WIP?